ibis/backends/duckdb/__init__.py

```diff
@@ -1797,3 +1903,30 @@ def _pyarrow_rbr(source, table_name, _conn, **_: Any):
     # Ensure the reader isn't marked as started, in case the name is
     # being overwritten.
     _conn._record_batch_readers_consumed[table_name] = False
+
+
+def _attach_name(path_or_url: str | Path) -> str:
+    """Return the name to use when attaching a database."""
+    url = urllib.parse.urlparse(str(path_or_url))
+    final = Path(url.path).stem
+    if not final:
+        raise ValueError(
+            f"Could not determine a name for the database from {path_or_url!r}. "
+            "Please provide a name explicitly using the `name` parameter."
+        )
+    return final
+
+
+for inp, expected in [
+    ("myddb", "myddb"),
+    ("myddb.duckdb", "myddb"),
+    ("myddb.temp.duckdb", "myddb.temp"),
+    ("myddb.temp.sqlite", "myddb.temp"),
+    (Path("myddb.duckdb"), "myddb"),
+    (Path("myddb"), "myddb"),
+    ("/root/myddb.duckdb", "myddb"),
+    ("myddb", "myddb"),
+    ("duckdb:///myddb.duckdb", "myddb"),
+    ("https://example.com/myddb.duckdb", "myddb"),
+]:
+    assert _attach_name(inp) == expected
```

Comment:
This should be in a test file. It's fine to import a private function in a test file.

Reply:
One reason is that we shouldn't need to run this code every time someone imports this module (indirectly or otherwise).

Reply:
I was trying to avoid importing a private function was my reasoning here. I figured the execution cost (only runs during first import, and will be CPU bound w/ no IO needed) would be so minimal that it was just easier to keep it here. Are you sure you still sure you want it in a test file?

Reply:
Ah, I see, that makes sense, thanks for the explanation

Reply:
I moved it to test_io.py